internal/backend/temporalclient/client.go

```diff
@@ -116,21 +100,45 @@ func New(cfg *Config, z *zap.Logger) (Client, error) {
 	return impl, nil
 }
 
+func (c *impl) startDevServer(ctx context.Context, cfg *Config, opts client.Options) error {
+	cfg.DevServer.ClientOptions = &opts
+
+	var err error
+	if c.srv, err = testsuite.StartDevServer(ctx, cfg.DevServer); err != nil {
+		return fmt.Errorf("start Temporal dev server: %w", err)
+	}
+	c.z.Info("started Temporal dev server", zap.String("address", c.srv.FrontendHostPort()))
+
+	c.client = c.srv.Client()
+
+	return nil
+}
+
 func (c *impl) Temporal() client.Client { return c.client }
 
 func (c *impl) Stop(context.Context) error {
 	close(c.done)
+
+	if c.client != nil {
```

Comment:
could c.client be nil?

Reply:
Technically yes - right at the beginning. Anyway, it's smarter to check rather than assume, in order to avoid a panic.